clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -1031,6 +1033,7 @@ public void run() {
                         } else if (heartbeat.pollTimeoutExpired(now)) {
                             // the poll timeout has expired, which means that the foreground thread has stalled
                             // in between calls to poll(), so we explicitly leave the group.
+                            rebalanceInProgress.compareAndSet(false, true);
```

Comment:
This flag is better set in LeaveGroupResponseHandler where we know whether the LeaveGroup request succeeded or not

Reply:
You have a point, I will make the change. :)